### PR TITLE
[JENKINS-42752] - Rename slaves to agents where possible

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Just addresses some technical debt.
Not all localizations are updated (e.g. Spanish and French need polishing). @amuniz was going to do the Spanish one.

https://issues.jenkins-ci.org/browse/JENKINS-42752

@reviewbybees

### CODE REVIEW

**Comment by ghost** [comment]
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

File: src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/help_de.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
 blurb=\
-  Startet einen Windows-Slave �ber einen <a href="http://en.wikipedia.org/wiki/Windows_Management_Instrumentation">\
-  Fernwartungsmechanismus</a>, der in Windows integriert ist. Geeignet f�r Windows-Slaves. \
-  Slaves m�ssen per IP vom Master-Knoten aus erreichbar sein.
+  Startet einen Windows-Agent \u00fcber einen <a href="http://en.wikipedia.org/wiki/Windows_Management_Instrumentation">\
+  Fernwartungsmechanismus</a>, der in Windows integriert ist. Geeignet f\u00fcr Windows-Agents. \
+  Agents m\u00fcssen per IP vom Master-Knoten aus erreichbar sein.
```
Comment (orrc) [inline src/main/resources/hudson/os/windows/ManagedWindowsServiceLauncher/help_de.properties:4]: Agents -> Agenten

File: src/main/resources/hudson/os/windows/Messages.properties [hunk 2]
```diff
@@ -22,8 +22,8 @@
 
 ManagedWindowsServiceLauncher.DisplayName=\
   Let Jenkins control this Windows slave as a Windows service
-ManagedWindowsServiceLauncher.DotNetRequired={0} [windows-slaves] .NET Framework 2.0 or later is required on this computer to run a Jenkins slave as a Windows service
-ManagedWindowsServiceLauncher.InstallingSlaveService={0} [windows-slaves] Installing the Jenkins slave service
+ManagedWindowsServiceLauncher.DotNetRequired={0} [windows-slaves] .NET Framework 2.0 or later is required on this computer to run a Jenkins agent as a Windows service
+ManagedWindowsServiceLauncher.InstallingSlaveService={0} [windows-slaves] Installing the Jenkins agent service
```
Comment (orrc) [inline src/main/resources/hudson/os/windows/Messages.properties:26]: Does `[windows-slaves]` have a special meaning here that can't be replaced?

Comment (oleg-nenashev) [reply src/main/resources/hudson/os/windows/Messages.properties:26]: It is a plugin artifact ID :(

Comment (oleg-nenashev) [reply src/main/resources/hudson/os/windows/Messages.properties:26]: But yes, we can change it if we want. I mean the text

Comment (orrc) [reply src/main/resources/hudson/os/windows/Messages.properties:26]: Ah, ok. I thought it was maybe a Windows service name or something.

But end users shouldn't ever really care about the artifact ID — the plugin name has been updated, which will be shown on plugins.jenkins.io and the Plugin Manager.  So it seems ok to replace?

Comment (oleg-nenashev) [reply src/main/resources/hudson/os/windows/Messages.properties:26]: Yep, I think so. It sneaks into the UI

**Comment by oleg-nenashev** [comment]
There are still some usages of the "slave" term internally, but not in the Web UI. I do not plan to proceed with further cleanup, but I will land this pull request so that some bits are integrated

**Pull request merged**